hat/assets/js/apps/Iaso/domains/forms/fields/hooks/useGetQueryBuildersFields.ts

```diff
@@ -1,41 +1,99 @@
-import { QueryBuilderFields } from 'bluesquare-components';
+import { QueryBuilderFields, useSafeIntl } from 'bluesquare-components';
 
 import { formatLabel } from '../../../instances/utils';
 
 import { PossibleFieldsForForm } from '../../hooks/useGetPossibleFields';
-import { FormDescriptor, PossibleField } from '../../types/forms';
+import { FieldType, FormDescriptor, PossibleField } from '../../types/forms';
 
 import { findDescriptorInChildren } from '../../../../utils';
 import { Field, iasoFields } from '../constants';
+import MESSAGES from '../../../workflows/messages';
+
+// existing mappings are referenced here: https://docs.openiaso.com/pages/dev/how_to/create_forms_for_entities/create_forms_for_entities.html#type-indicators
+const calculateMapping = [
+    {
+        suffix: '__int__',
+        type: 'integer',
+    },
+    {
+        suffix: '__integer__',
+        type: 'integer',
+    },
+    {
+        suffix: '__long__',
+        type: 'decimal',
+    },
+    {
+        suffix: '__decimal__',
+        type: 'decimal',
+    },
+    {
+        suffix: '__double__',
+        type: 'decimal',
+    },
+    {
+        suffix: '__bool__',
+        type: 'boolean',
+    },
+    {
+        suffix: '__boolean__',
+        type: 'boolean',
+    },
+    {
+        suffix: '__date__',
+        type: 'date',
+    },
+    {
+        suffix: '__time__',
+        type: 'time',
+    },
+    {
+        suffix: '__date_time__',
+        type: 'integer',
+    },
+    {
+        suffix: '__datetime__',
+        type: '__date_time__',
+    },
+];
 
 export const useGetQueryBuildersFields = (
     formDescriptors?: FormDescriptor[],
     possibleFields?: PossibleField[],
     configFields: Field[] = iasoFields,
 ): QueryBuilderFields => {
+    const { formatMessage } = useSafeIntl();
     if (!possibleFields || !formDescriptors) return {};
     // you can fields examples here: https://codesandbox.io/s/github/ukrbublik/react-awesome-query-builder/tree/master/sandbox?file=/src/demo/config.tsx:1444-1464
     const fields: QueryBuilderFields = {};
     possibleFields.forEach(field => {
+        const fieldCopy = { ...field };
+        const mapping = calculateMapping.find(
+            m => field.name.endsWith(m.suffix) && field.type === 'calculate',
+        );
```

Comment:
Nitpicking here, mostly, but I think you should rewrite it this way to make it more efficient:
```suggestion
        const mapping =  field.type === 'calculate' ? 
            mapping = calculateMapping.find(
                m => field.name.endsWith(m.suffix),
            ) : null;
```
This will avoid looping on all mappings for fields that are not `calculate`s.